eZ/Publish/Core/FieldType/DateAndTime/Type.php

```diff
@@ -246,19 +258,23 @@ public function validateFieldSettings($fieldSettings)
                         break;
                     case 'dateInterval':
                         if (isset($value)) {
-                            if (isset($fieldSettings['defaultType']) &&
-                                $fieldSettings['defaultType'] !== self::DEFAULT_CURRENT_DATE_ADJUSTED) {
-                                $validationErrors[] = new ValidationError(
-                                    "Setting '%setting%' can be used only when setting '%defaultType%' is set to '%DEFAULT_CURRENT_DATE_ADJUSTED%'",
-                                    null,
-                                    array(
-                                        'setting' => $name,
-                                        'defaultType' => 'defaultType',
-                                        'DEFAULT_CURRENT_DATE_ADJUSTED' => 'DEFAULT_CURRENT_DATE_ADJUSTED',
-                                    ),
-                                    "[$name]"
-                                );
-                            } elseif (!($value instanceof \DateInterval)) {
+                            if ($value instanceof \DateInterval) {
+                                // String conversion of $value, because DateInterval objects cannot be compared directly
+                                if (isset($fieldSettings['defaultType']) &&
+                                    $fieldSettings['defaultType'] !== self::DEFAULT_CURRENT_DATE_ADJUSTED &&
+                                    $value->format('%y%m%d%h%i%s') !== '000000') {
+                                    $validationErrors[] = new ValidationError(
+                                        "Setting '%setting%' can be used only when setting '%defaultType%' is set to '%DEFAULT_CURRENT_DATE_ADJUSTED%'",
+                                        null,
+                                        array(
+                                            'setting' => $this->translator->trans('field_definition.ezdatetime.date_interval', [], 'ezrepoforms_content_type'),
+                                            'defaultType' => $this->translator->trans('field_definition.ezdatetime.default_type', [], 'ezrepoforms_content_type'),
+                                            'DEFAULT_CURRENT_DATE_ADJUSTED' => $this->translator->trans('field_definition.ezdatetime.default_type_adjusted', [], 'ezrepoforms_content_type'),
+                                        ),
```

Comment:
You should not use Symfony services in API related stuff. `ValidationError` is already translatable. Here you just need to assign the values. Why do you need translation for this?